.github/workflows/sdk-generation.yaml

```diff
@@ -391,6 +399,32 @@ jobs:
           footer: "Linked Repo <{repo_url}|{repo}> | <{run_url}|View Run>"
         env:
           SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+  compile-ruby:
+    if: ${{ needs.generate.outputs.ruby_regenerated == 'true' }}
+    name: Compile Ruby SDK
+    runs-on: ubuntu-latest
+    needs: generate
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.generate.outputs.branch_name }}
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@ec02537da5712d66d4d50a0f33b7eb52773b5ed1
+        with:
+          ruby-version: "3.1"
+      - run: bundle install
```

Comment:
generally the steps here we will want to mirror the steps in the `compile.js` file from the templates ie `gem build && bundle install && rake rubocop:autocorrect` as we want this step to fail if the generator outputs invalid ruby